src/store/reducers.ts

```diff
@@ -56,9 +63,47 @@ export function relayer(state: RelayerState = initialRelayerState, action: RootA
     return state;
 }
 
+export function stepsModal(state: StepsModalState = initialStepsModal, action: RootAction): StepsModalState {
+    switch (action.type) {
+        case getType(actions.setStepsModalVisibility):
+            return { ...state, isVisible: action.payload };
+        case getType(actions.setStepsModalDoneSteps):
+            return { ...state, doneSteps: action.payload };
+        case getType(actions.setStepsModalPendingSteps):
+            return { ...state, pendingSteps: action.payload };
+        case getType(actions.setStepsModalCurrentStep):
+            return { ...state, currentStep: action.payload };
+        case getType(actions.stepsModalAdvanceStep):
+            const { doneSteps, currentStep, pendingSteps } = state;
+            if (pendingSteps.length === 0 && currentStep !== null) {
+                return {
+                    ...state,
+                    doneSteps: doneSteps.concat([currentStep as Step]),
+                    currentStep: null,
+                };
+            } else if (pendingSteps.length > 1) {
```

Comment:
I *think* you can make this just the `else` and remove the block below. It seems to me they do the same thing when `pendingSteps.length === 1`. Am I missing something?